### DISCUSSION TITLE
Tagbar included in default setup

### DISCUSSION THREAD

**Discussion by jasoncmcg** [LunarVim/LunarVim#2838] [General]
### Feature Description

As an alternative to a minimap, **tagbar** offers an excellent option as it includes a summary overview of the code with a bookmark link type of ability to jump to the relevant part of your code. 

LunarVim is awesome and is only missing this one aspect... that I can tell. (The aspect of a minimap type of option.)

I would guess that a response might be that it requires something external to be installed. However, the icons required me to install a font and then also configure konsole to use the font.

### Describe the alternatives you have considered

I thought there was maybe something with tree-sitter, but that appears to only be doing keyword highlighting...

**Comment by kobzar** [comment]
I think this is more personal moment. 
For my opinion, this plugin isn’t effective and need just for beautiful screen. 
If file is not big - neoscroll is better choice. If file really big - i use marks. On both variants we dont spent time to left work. 
The same i can say about minimap. 
All this beauty’s  steal our working place and unfocusing our attention. 
So, thats why this plugin like others(colorizer, save work, surround e.t.c) is leave for user choice. 

I mean thats LunarVim is good for the start, but if you need more - you can.
But tagbar need only for begginers.

**Reply by jasoncmcg** [reply]
Thank you for the feedback, I appreciate it. I feel like marks are a more living or dynamic feature. Whereas tags are more like an autogenerated table of contents that give a high level preview of the entire code file. I would equate it more to NvimTree on the inside of the file, instead of on the list of files. In any case, I have added it in the designated section for user plugins, as you mentioned. Thank you.

**Comment by andreacfromtheapp** [comment]
Hi :)

I've just found out about this discussion. I was going to add a feature request/PR, but probably this is a better place :)

I was soooooooo frustrated with minimap, its conflicts, and its bugs, that I seeked for an alternative for days. I finally found a GREAT ONE called [mini.map](https://github.com/echasnovski/mini.nvim/blob/main/readmes/mini-map.md) :)

It works ridiculously well, it's extremley fast and lightweight and it can double as a sidebar too! Tbh it's so amazing it should be in core imho.

My Lua is very limited (like nothing at all), and I'm sure the LunarVim would make the most of it. For example, most things require some Lua, and so far I cannot make it [open by default](https://github.com/echasnovski/mini.nvim/blob/89c77d220f546782f1da12fe451c6a56c479ffd4/doc/mini-map.txt#L59-L70). 

That said, I was so excited that I wanted to share and suggest it! Here is how I'm currently using it (still working on the autocmd to open it by default):

```
  {
    "echasnovski/mini.map",
    branch = "stable",
    config = function()
      require('mini.map').setup()
      local map = require('mini.map')
      map.setup({
        integrations = {
          map.gen_integration.builtin_search(),
          map.gen_integration.gitsigns({
            add = 'GitSignsAdd',
            change = 'GitSignsChange',
            delete = 'GitSignsDelete',
          }),
          map.gen_integration.diagnostic({
            error = 'DiagnosticFloatingError',
            warn  = 'DiagnosticFloatingWarn',
            info  = 'DiagnosticFloatingInfo',
            hint  = 'DiagnosticFloatingHint',
          }),
        },
        symbols = {
          encode = map.gen_encode_symbols.dot('4x2'),
          scroll_line = '➽',
          scroll_view = '┃',
        },
        window = {
          side = 'right', -- Side to stick ('left' or 'right')
          width = 17, -- Total width
          winblend = 25, -- Value of 'winblend' option
          focusable = true, -- Whether window is focusable in normal way (with `wincmd` or mouse)
          show_integration_count = false, -- Whether to show count of multiple integration highlights
        },
      })
    end
  },
```

Here's it's how it looks with the above:

<img width="1920" alt="Screenshot 2022-11-25 at 7 52 39 PM" src="https://user-images.githubusercontent.com/3269984/204041814-3309e541-f052-47b7-9b34-f6b92385907f.png">

Looking forward to know what LunarVim thinks of it and to being included and properly configured!

Cheers

**Comment by kobzar** [comment]
Yep, the map plugin isn’t work. I tried it half year ago. And as you can see - no one issue was opened about it. So, minimap plugin don’t uses over of 99% of users. 
but thanks for the attention.

**Reply by andreacfromtheapp** [reply]
> Yep, the map plugin isn’t work. I tried it half year ago. And as you can see - no one issue was opened about it. So, minimap plugin don’t uses over of 99% of users. but thanks for the attention.

hi @kobzar , the plugin **is** working. It was just my lua noobness... Anyway, if nobody cares, that's fine.

However, I think and IDE should really have a minimap, imho. In the unlikely case you change your mind (or someone comes looking for one), I'm leaving here my current config.  -- EDIT: proper config and working mini.map with sensible defaults and amazing funciontality!

- Packer:

```
...
  {
    "echasnovski/mini.map",
    branch = "stable",
    config = function()
      require('mini.map').setup()
      local map = require('mini.map')
      map.setup({
        integrations = {
          map.gen_integration.builtin_search(),
          map.gen_integration.diagnostic({
            error = 'DiagnosticFloatingError',
            warn  = 'DiagnosticFloatingWarn',
            info  = 'DiagnosticFloatingInfo',
            hint  = 'DiagnosticFloatingHint',
          }),
        },
        symbols = {
          encode = map.gen_encode_symbols.dot('4x2'),
        },
        window = {
          side = 'right',
          width = 20, -- 1 for a pure scrollbar :)
          winblend = 15,
          show_integration_count = false,
        },
      })
    end
  },
...
```

- autocommands:

```
-- [[ AUTOCOMMANDS ]]

lvim.autocommands = {
  ...
  {
    "BufEnter",
    {
      desc = "Open mini.map and exclude some filtypes",
      pattern = { "*" },
      callback = function()
        local exclude_ft = {
          "",
          "nofile",
          "nowrite",
          "quickfix",
          "prompt",
          "terminal",
          "toggleterm",
          "startify",
          "dashboard",
          "alpha",
          "netrw",
          "NvimTree",
          "packer",
          "telescope",
          "lazygit",
          "bufferline",
        }

        local map = require('mini.map')
        if vim.o.buftype == "" and not vim.tbl_contains(exclude_ft, vim.o.filetype) then
          map.open()
        elseif vim.o.buftype == "nofile" and vim.o.filetype == "minimap" then
          return
        else
          vim.b.minimap_disable = true
          map.close()
        end
      end,
    },
  },
  ...
}

```

cheers!

**Comment by andreacfromtheapp** [comment]
proposed a PR to the extra plugins doc. discord chat said it was welcome: https://github.com/LunarVim/lunarvim.org/pull/306
